pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -151,6 +152,27 @@ func updateVitessShard(key client.ObjectKey, vts *planetscalev2.VitessShard, vtk
 	// Update labels, but ignore existing ones we don't set.
 	update.Labels(&vts.Labels, newShard.Labels)
 
+	// Remove old annotations that shouldn't be there that we injected previously.
+	differentAnnotations := differentKeys(newShard.Annotations, vts.Spec.Annotations)
```

Comment:
These should be reversed, right? Also, I think we should compare against `newShard.Spec.Annotations` so we don't get confused if later there is an extra annotation in `newShard.Annotations` that is not part of the set that comes from `newShard.Spec.Annotations`.

```suggestion
	differentAnnotations := differentKeys(vts.Spec.Annotations, newShard.Spec.Annotations)
```

Reply:
Yep, good catch! https://github.com/planetscale/vitess-operator/pull/51/commits/e12e40a8d7d1f5b6127b6a00b1f8bbd4481cb875